### DISCUSSION TITLE
Chain Lightning problem

### DISCUSSION THREAD

**Discussion by N3ur0funk-Infinity** [diasurgical/DevilutionX#6253] [Q&A] [answered]
Hi everyone!
When there are many enemies and I cast a few times Chain Lightning the game lags for 1-2 seconds and fps drops from 150-140 to 90-80 fps. I've never had this problem playing vanilla Diablo or Hellfire. I have a 1440p 165Hz monitor but I play the game at 1280x720 resolution for better experience. At higher resolutions everything is way too small for me.

My PC specs:
i7 8700k
MSI GeForce RTX 3080 Suprim X 10GB
16GB RAM Kingston clocked to 3200Mhz with XMP (2x8GB)
Asus Maximus X Hero
Game is installed on HDD (2TB WD Black) but my Windows 10 runs on ADATA M2 NVme 512GB (SX8200PNP)

Any ideas what is causing the problem? I play mostly on Loopback and speed is set to Faster.
Could it be audio sample rate set to 48000 + buffer size 5120 + resampling quality 10?

**Comment by AJenbo** [comment, marked as answer]
Resampling quality 10 is probably what's causing that, please leave it around 3 unless you are having some issues.

**Reply by N3ur0funk-Infinity** [reply]
Thanks a lot, it helped!

**Reply by julealgon** [reply]
@AJenbo should we do something to limit the number of simultaneous sounds (I assume this is why the problem is happening)? I remember even in vanilla that the sound becomes this garbled mess when chain lightning produced a lot of bolts.

**Reply by N3ur0funk-Infinity** [reply]
@julealgon No, it's really ok. It's better than in vanilla or Hellfire. At least it feels so, it's not glitchy or something. No need to modify it! :)

**Reply by AJenbo** [reply]
The game already applies some limits on duplicate sounds being played. More so I think we should limit the allowed ranges for sound quality, a lot of people set it to the highest number expecting something good

**Comment by N3ur0funk-Infinity** [comment]
> The game already applies some limits on duplicate sounds being played. More so I think we should limit the allowed ranges for sound quality, a lot of people set it to the highest number expecting something good

To be honest, that's exactly what I thought so I set it to 10, expecting wonders. If there are more people doing the same mistake as I did, then it's worth considering to limit it.
But, can I ask - what's the point of this option? Is it purely for listening to the soundtrack? Pardon me, it's none of my business, but I'm just curious. You don't have to answer me. You've helped me more than enough and I'm really grateful!

**Reply by StephenCWills** [reply]
It's configuration for the audio resampler. It controls the filters that will be applied to the audio during resampling.

**Reply by AJenbo** [reply]
The firewall is only 11khz and needs to be resampled to be able to mix it with the other sounds of the game. The rest is 22khz so do not need to be resampled unless you set the output to something other then 22khz. If you do so all audio will need to be resampled.

A higher buffer size will add additional delay to the audio in case you have a slow CPU that is unable to process audio frequently, at the cost of extra ram (and a delay in the sound).

If you set a higher sample rate all audio will need to be resampled by the application. This would normally not be preferred unless the resample for your sound card is particularly bad (if it's native sample rate does not match the output of the application it will again need to be resampled to be mixed with desktop sounds and converted to the analog audible audio)

**Reply by N3ur0funk-Infinity** [reply]
@StephenCWills Nice! Thank you for your fast reply! I love this mod, it's simply the best. It keeps all the the best of the original and brings so many QoL upgrades.
Have a nice day to all of you!

@AJenbo This is a great answer, thank you very much! You explained the problem perfectly. I admire you guys for all the work you put into this project! It's the best, because the right people work on it. You've brought this classic to life the best possible way! 🙏
